### ISSUE TITLE
How to send caption with each media while creating Event ?

### ISSUE THREAD

**Issue by inder243** [brianhorakh/vita#17]
We have no use of the same in the application.Please clarify the use of the caption.

**Comment by SeanGuerrero** [comment]
![picturecaptions](https://cloud.githubusercontent.com/assets/19561916/15732585/44d5eb20-2833-11e6-9e84-55e6bee0345f.png)

Each of the individual pictures or videos that you upload into a created event in 10D, needs to be able to have separate captions.  This is where a user can choose to describe their personal account of what is happening in the pic/video.  The captions will be visible from the Pic/Video Info screen seen in 9A.  9A is only accessible from a full frame pic or video.

Every pic seen in grid format of an event, seen in 8B, will possibly have a separate caption to describe what is happening.

**Comment by brianhorakh** [comment]
Aren't the caption/title the same thing?  (use 'title' parameter in the EventCreate api call)
Sean - can you please confirm?

**Comment by SeanGuerrero** [comment]
There are two things that are being discussed.  There is a Title that goes with the overall event being created.  Then there is an option to include a Caption/Description to each individual pic or video being uploaded into that event.

**Comment by brianhorakh** [comment]
Sean - at this moment EventMediaAppend does not have support for a 'caption' it's just media, which can then be annotated by the user using a comment.
I thought a caption would be added as an Annotation/comment to the media.
Are you requesting that I add 'caption' as an input parameter to a piece of media?

**Comment by SeanGuerrero** [comment]
Yes, add 'caption' as an input parameter for adding a pic/video to or with an event.

**Comment by SeanGuerrero** [comment]
Please refer to the 10D image above.  There are separate boxes for event title and media caption.

**Comment by brianhorakh** [comment]
Alright, I've added a 'caption' and 'title' parameter to EventMediaAppend,  I've also added an EventMediaUpdate api call since that will be required to make changes to the title, caption and/or switch out media (which I would prefer we didn't actually support since that's a bit hairy with file caching -- needs testing)

**Comment by inder243** [comment]
http://www.vitatheapp.com/vita-api/EventMediaUpdate?evid=D3347DC6-F71B-11E5-91E5-A7AA58C3F44C&authid=4F45325C-F71C-11E5-A06D-A7AA58C3F44C&mediaid=327&uid=101&title=updatedtitle&caption=updatedcaption

Response : {"success":false,"error_msg":"internal server error Undefined subroutine &VITA::update called at /httpd/vita/perllib/VITA.pm line 1595.\n","now":"2016-06-06 05:41:02","error_id":666}

**Comment by SeanGuerrero** [comment]
Brian, is this an error for you to check out too?

**Comment by brianhorakh** [comment]
Thanks for the heads up Sean, but the issue should already be corrected.

**Comment by inder243** [comment]
Thanks for adding "caption" param in  EventMediaAppend api, but we need caption of each media in eventlist api also, so that we can show it corresponding to each media (refer above screen 9A)

**Comment by inder243** [comment]
Can you please clarify on which view click, we have to show screen 9A(mentioned above). Also specify click event of buttons and views on 9A screen.

**Comment by brianhorakh** [comment]
"caption" has been added to the EventList response.

**Comment by SeanGuerrero** [comment]
Vakul, The functionality for 9A screen and all recent UI changes can be found in this document.

[Functionality for New Vita Update 5-11-16.docx](https://github.com/brianhorakh/vita/files/317536/Functionality.for.New.Vita.Update.5-11-16.docx)

**Comment by inder243** [comment]
ok will add and let you know if facing any issue
